app/models/order_cycle.rb

```diff
@@ -24,7 +24,7 @@ class OrderCycle < ApplicationRecord
   has_many :distributors, -> { distinct }, source: :receiver, through: :cached_outgoing_exchanges
   has_many :order_cycle_schedules
   has_many :schedules, through: :order_cycle_schedules
-  has_many :order_cycle_shipping_methods
+  has_many :order_cycle_shipping_methods, dependent: :destroy
```

Comment:
I see. And I was wondering if this is the best approach. A quick search lead me to this: https://stackoverflow.com/questions/40016670/should-we-write-dependent-destroy-on-a-join-table-model

And I like that solution. It ensures the database integrity and is less code for us here.